components/time/src/provider/mod.rs

```diff
@@ -65,6 +64,21 @@ pub const MARKERS: &[DataMarkerInfo] = &[
 
 const SECONDS_TO_EIGHTS_OF_HOURS: i32 = 60 * 60 / 8;
 
+/// A time zone variant used to identify a display name in CLDR.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+#[zerovec::make_ule(TimeZoneVariantULE)]
+#[repr(u8)]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize, databake::Bake))]
+#[cfg_attr(feature = "datagen", databake(path = icu_time::provider))]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[non_exhaustive]
+pub enum TimeZoneVariant {
+    /// The variant corresponding to `"standard"` in CLDR.
+    Standard = 0,
+    /// The variant corresponding to `"daylight"` in CLDR.
+    Daylight = 1,
+}
```

Comment:
I think we should retain the notice that was there previously

```suggestion
pub enum TimeZoneVariant {
    /// The variant corresponding to `"standard"` in CLDR.
    ///
    /// The semantics vary from time zone to time zone. The time zone display
    /// name of this variant may or may not be called "Standard Time".
    Standard = 0,
    /// The variant corresponding to `"daylight"` in CLDR.
    ///
    /// The semantics vary from time zone to time zone. The time zone display
    /// name of this variant may or may not be called "Daylight Time".
    Daylight = 1,
}
```